astroquery/mast/utils.py

```diff
@@ -155,22 +155,48 @@ def parse_input_location(coordinates=None, objectname=None):
 
 def mast_relative_path(mast_uri):
     """
-    Given a MAST dataURI, return the associated relative path.
+    Given one or more MAST dataURI(s), return the associated relative path(s).
 
     Parameters
     ----------
-    mast_uri : str
-        The MAST uri.
+    mast_uri : str, list of str
+        The MAST uri(s).
 
     Returns
     -------
-    response : str
-        The associated relative path.
+    response : str, list of str
+        The associated relative path(s).
     """
-
-    response = _simple_request("https://mast.stsci.edu/api/v0.1/path_lookup/",
-                               {"uri": mast_uri})
-    result = response.json()
-    uri_result = result.get(mast_uri)
-
-    return uri_result["path"]
+    if isinstance(mast_uri, str):
+        uri_list = [("uri", mast_uri)]
+    else:  # mast_uri parameter is a list
+        uri_list = [("uri", uri) for uri in mast_uri]
+
+    # Split the list into chunks of 50 URIs; this is necessary
+    # to avoid "414 Client Error: Request-URI Too Large".
+    uri_list_chunks = list(_split_list_into_chunks(uri_list, chunk_size=50))
+
+    result = []
+    for chunk in uri_list_chunks:
+        response = _simple_request("https://mast.stsci.edu/api/v0.1/path_lookup/",
+                                   {"uri": chunk})
+        json_response = response.json()
+        for uri in chunk:
+            path = json_response.get(uri[1])["path"]
```

Comment:
I don't see what's expected to be in `uri` here, could you explain it a bit, so I see why the second element?